File: API/App.Model/Entities/Coach.cs
using App.Model.Entities.Common;
using App.Model.Enums;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace App.Model.Entities
{
    public class Coach : AuditableEntity
    {
        [Required]
        public Guid UserId { get; set; }
        public User User { get; set; }

        [Required]
        public DateTime BirthDate { get; set; }

        [Required]
        public string Country { get; set; }

        [Required]
        public Gender Gender { get; set; }
        public DateTime? StartedWorking { get; set; }
        public DateTime? FinishedWorking { get; set; }
        public ICollection<Match> Matches { get; set; }
        public ICollection<Training> Trainings { get; set; }
        public ICollection<Team> Teams { get; set; }
        public ICollection<CoachCard> Cards { get; set; }
    }
}

File: WebUI/app-frontend/src/app/models/coach/coach.ts
import { Gender } from "src/app/enums/gender";
import { SimpleTeam } from "../team/simple-team.model";

export class Coach {
    public id : string;
    public isActive : boolean;
    public email: string;
    public name : string;
    public middleName : string;
    public surname : string;
    public birthDate : Date;
    public country : string;
    public gender : Gender;
    public startedWorking : Date;
    public finishedWorking : Date;
    public teams : Array<SimpleTeam>;
}
